docs/aws/audit/redshiftmonitoring/rules/default_master_username.mdx

```diff
@@ -24,6 +24,249 @@ PCIDSS
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the use of the default master username in Amazon Redshift using AWS CLI, follow these steps:
+
+1. **Create a New Redshift Cluster with a Custom Master Username:**
+   When creating a new Redshift cluster, specify a custom master username using the `--master-username` parameter.
+
+   ```sh
+   aws redshift create-cluster \
+       --cluster-identifier my-cluster \
+       --node-type dc2.large \
+       --master-username customadmin \
+       --master-user-password MySecurePassword123 \
+       --cluster-type single-node
+   ```
+
+2. **Modify an Existing Redshift Cluster to Change the Master Username:**
+   Note that AWS does not allow changing the master username of an existing cluster directly. Instead, you need to create a new cluster with the desired username and migrate your data.
+
+3. **Use AWS CloudFormation to Enforce Custom Master Username:**
+   Define a CloudFormation template that specifies a custom master username. This ensures that any Redshift cluster created using this template will not use the default master username.
+
+   ```yaml
+   Resources:
+     MyRedshiftCluster:
+       Type: "AWS::Redshift::Cluster"
+       Properties:
+         ClusterIdentifier: "my-cluster"
+         NodeType: "dc2.large"
+         MasterUsername: "customadmin"
+         MasterUserPassword: "MySecurePassword123"
+         ClusterType: "single-node"
+   ```
+
+   Deploy the CloudFormation stack using AWS CLI:
+
+   ```sh
+   aws cloudformation create-stack \
+       --stack-name my-redshift-stack \
+       --template-body file://my-redshift-template.yaml
+   ```
+
+4. **Automate Cluster Creation with a Custom Username Using a Python Script:**
+   Use the AWS SDK for Python (Boto3) to automate the creation of a Redshift cluster with a custom master username.
+
+   ```python
+   import boto3
+
+   redshift = boto3.client('redshift')
+
+   response = redshift.create_cluster(
+       ClusterIdentifier='my-cluster',
+       NodeType='dc2.large',
+       MasterUsername='customadmin',
+       MasterUserPassword='MySecurePassword123',
+       ClusterType='single-node'
+   )
+
+   print(response)
+   ```
+
+   Run the Python script to create the cluster:
+
+   ```sh
+   python create_redshift_cluster.py
+   ```
+
+By following these steps, you can ensure that your Redshift clusters do not use the default master username, thereby enhancing the security of your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the use of the default master username in Amazon Redshift clusters using Python scripts, you can use the AWS SDK for Python (Boto3). Below are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>